runbotics/runbotics-scheduler/src/websocket/gateway/bot.gateway.ts

```diff
@@ -76,10 +74,19 @@ export class BotWebSocketGateway implements OnGatewayDisconnect, OnGatewayConnec
         @ConnectedSocket() socket: BotAuthSocket,
         @MessageBody() processInstanceEvent: IProcessInstanceEvent,
     ) {
-        const installationId = socket.bot.installationId;
-        this.logger.log(`=> Updating process-instance-event (${processInstanceEvent.executionId}) by bot (${installationId}) | step: ${processInstanceEvent.step}, status: ${processInstanceEvent.status}`);
-        await this.botProcessEventService.updateProcessInstanceEvent(processInstanceEvent, socket.bot);
-        this.logger.log(`<= Success: process-instance-event (${processInstanceEvent.executionId}) updated by bot (${installationId}) | step: ${processInstanceEvent.step}, status: ${processInstanceEvent.status}`);
+        const updateProcessInstanceEvent = async () => {
+            const installationId = socket.bot.installationId;
+            this.logger.log(`=> Updating process-instance-event (${processInstanceEvent.executionId}) by bot (${installationId}) | step: ${processInstanceEvent.step}, status: ${processInstanceEvent.status}`);
+            await this.botProcessEventService.updateProcessInstanceEvent(processInstanceEvent, socket.bot);
+            this.logger.log(`<= Success: process-instance-event (${processInstanceEvent.executionId}) updated by bot (${installationId}) | step: ${processInstanceEvent.step}, status: ${processInstanceEvent.status}`);
+        };
+
```

Comment:
imo this fn outside of method as a new private method so it will be created once, not every message delivered